src/main/java/org/broadinstitute/hellbender/utils/codecs/ProgressReportingDelegatingCodec.java

```diff
@@ -0,0 +1,125 @@
+package org.broadinstitute.hellbender.utils.codecs;
+
+import htsjdk.samtools.util.LocationAware;
+import htsjdk.tribble.Feature;
+import htsjdk.tribble.FeatureCodec;
+import htsjdk.tribble.FeatureCodecHeader;
+import org.broadinstitute.hellbender.engine.ProgressMeter;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * This class is useful when we want to report progress when indexing. The indexing code is in htsjdk and not available to us directly.
+ * The workaround is to make a special decorator 'codec' that gets called on every decoded Feature and this can use used to track progress.
+ * The codec delegates all calls and reports progress as it goes.
+ */
+public final class ProgressReportingDelegatingCodec<A extends Feature, B> implements FeatureCodec<A, B> {
+    private final FeatureCodec<A, B> delegatee;
+
+    private final ProgressMeter pm;
+
+    //Note: this default constructor is needed for the FeatureManager when it loads codecs.
+    @SuppressWarnings("unused")
+    public ProgressReportingDelegatingCodec(){
+        delegatee = null;
+        pm = null;
+    }
+
+    public ProgressReportingDelegatingCodec(final FeatureCodec<A, B> delegatee, final double secondsBetweenUpdates){
+        if ( secondsBetweenUpdates <= 0.0 ) {
+            throw new IllegalArgumentException("secondsBetweenUpdates must be > 0.0");
+        }
+        this.delegatee = delegatee;
+        this.pm = new ProgressMeter(secondsBetweenUpdates);
+    }
+
+    @Override
+    public Feature decodeLoc(final B b) throws IOException {
+        if (delegatee == null) {
+            throw new IllegalStateException("this progress monitor cannot be used without a delegatee.");
```

Comment:
replace all instances of "this progress monitor" in the error messages in this class with "this codec"